src/os.rs

```diff
@@ -1,71 +1,69 @@
 #![no_std] // no standard library
 #![no_main]
+#![feature(custom_test_frameworks, abi_x86_interrupt)]
+#![test_runner(os_core::test_framework::test_runner)]
+#![reexport_test_harness_main = "test_main"]
 // no entry point
 
 // #################################################
 // #   This produces a runnable binary of the OS   #
 // #################################################
 
-// configures our test framework environment
-#![feature(custom_test_frameworks, abi_x86_interrupt)]
-#![test_runner(os_core::test_framework::test_runner)]
-#![reexport_test_harness_main = "test_main"]
-
+use os_core::vga::vga_core::{TextDrawable, PlainDrawable};
+use bootloader::{BootInfo, boot_info::FrameBuffer, entry_point};
+use os_core::vga::{vga_buffer::VGADeviceFactory, vga_core::Clearable, vga_color};
 use core::panic::PanicInfo;
 
-use os_core::{hlt_loop, interface::VGA_TEXT_BUFFER_INTERFACE, print, vga::text::CursorMode};
-
-#[no_mangle]
-// entry point of the program
-pub extern "C" fn _start() -> ! {
-    // initialisation steps of the OS
-    // ! DO NOT REMOVE!
-    os_core::init();
+use os_core::hlt_loop;
 
-    // `cargo test` entry point.
-    // ! DO NOT REMOVE!
+entry_point!(kernel);
+pub fn kernel(boot_info: &'static mut BootInfo) -> ! {
     #[cfg(test)]
-    test_main();
+    kernel_test(boot_info);
+    #[cfg(not(test))]
+    kernel_main(boot_info);
+    hlt_loop();
+}
 
-    // ########################################
-    // # _start() actual entry on `cargo run` #
-    // ########################################
+pub fn kernel_main(boot_info: &'static mut BootInfo) {
+    
+    let framebuffer_pointer: *mut FrameBuffer = boot_info.framebuffer.as_mut().unwrap();
 
-    VGA_TEXT_BUFFER_INTERFACE.lock().set_pos(12, 33);
-    VGA_TEXT_BUFFER_INTERFACE
-        .lock()
-        .set_cursor(CursorMode::Block);
-    print!("Hello, World!");
+    let os_framebuffer = unsafe { framebuffer_pointer.as_mut().unwrap() };
+    os_core::init(os_framebuffer);
 
-    hlt_loop();
-}
+    let usable_framebuffer = unsafe { framebuffer_pointer.as_mut().unwrap() };
+
+
+    let mut device = VGADeviceFactory::from_buffer(usable_framebuffer);
```

Comment:
would it make sense to put this "default device" somewhere as a global constant?

Reply:
Will do that in next PRs, as we do not have anything using it at the moment (only the stuff randomly drawn on the screen etc.)